CMakeLists.txt

```diff
@@ -39,6 +39,33 @@ project(
     LANGUAGES
         CXX
 )
+
+# Adjust compiler and linker flags needed by this project
+if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+    find_library(LIBC++ c++)
+    find_library(LIBC++ABI c++abi)
+    if(NOT LIBC++ OR NOT LIBC++ABI)
+        message(
+            FATAL_ERROR
+            "libc++ or libc++abi not found. Make sure you have libc++ and libc++abi installed and in your PATH"
+        )
+    endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
+else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsized-deallocation")
+endif()
+
+set(CMAKE_CXX_FLAGS_RELEASE "-O3")
+set(CMAKE_CXX_FLAGS_DEBUG "-O0 -g -DDEBUG")
```

Comment:
Do we know whether all the code tests this define only for existence and not for value?

Reply:
```
find . -name "*.h" -exec grep -H "DEBUG" {} \; | grep "#if"
find . -name "*.hpp" -exec grep -H "DEBUG" {} \; | grep "#if"
find . -name "*.cpp" -exec grep -H "DEBUG" {} \; | grep "#if"
```
Did not see any cases where the value was checked, but the output is quite noisy.